src/fondant/core/manifest.py

```diff
@@ -238,11 +238,76 @@ def remove_field(self, name: str) -> None:
 
         del self._specification["fields"][name]
 
+    @staticmethod
+    def _handle_custom_produces(
+        evolved_manifest: "Manifest",
+        produces: t.Optional[t.Union[t.Dict[str, str], t.Dict[str, Field]]],
+        component_spec: ComponentSpec,
+    ) -> t.Mapping[str, Field]:
```

Comment:
Can you add a docstring with a high level description of what is happening here?

If I understand correctly, you support a partial `produces` definition? So the user doesn't have to overwrite the `produces section completely?

Reply:
yes indeed, by default we're basing ourselves on the component specs and it can be optionally defined for some fields for renaming (for non-generic fields)